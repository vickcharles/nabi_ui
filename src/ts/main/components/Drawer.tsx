import * as React from 'react';
import { Link } from 'react-router-dom';

import Drawer from 'material-ui/Drawer';
import { MenuItem } from 'material-ui/Menu';
import IconButton from 'material-ui/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Divider from 'material-ui/Divider';

import { nonRegisteredUserMenuItems } from '../model';

interface DrawerMenuState {
  isDrawerOpen: boolean;
}

export default class DrawerMenu extends React.Component<{}, DrawerMenuState>  {
  constructor(props: {}) {
    super(props);
    this.state = {isDrawerOpen: false};
  }

  handleToggle = () => this.setState({isDrawerOpen: !this.state.isDrawerOpen});
  handleClose = () => this.setState({isDrawerOpen: false});

  render() {
    const menuItems: any = [];

    for (const [key, item] of Object.entries(nonRegisteredUserMenuItems)) {
      menuItems.push(
        <Link key={key} to={item.url}>
          <MenuItem className={item.className} onClick={this.handleClose}>
            {item.label}
          </MenuItem>
          {item.divider && <Divider />}
        </Link>
      );
    }

    return (
      <div>
        <IconButton onClick={this.handleToggle} id="menu" aria-label="Menu">
          <MenuIcon />
        </IconButton>
        <Drawer
          onClick={this.handleClose}
          open={this.state.isDrawerOpen}
        >
          {menuItems}
        </Drawer>
      </div>
    );
  }
}
