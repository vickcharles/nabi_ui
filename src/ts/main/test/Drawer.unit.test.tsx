import * as React from 'react';
import Drawer from '../components/Drawer';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';

describe('Drawer', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <Drawer />
      </MemoryRouter>
    ).find(Drawer).dive();
  });

  it('Renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
