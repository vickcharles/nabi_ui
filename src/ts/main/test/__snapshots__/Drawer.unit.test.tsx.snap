// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Renders the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DrawerMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(IconButton)
          aria-label="Menu"
          id="menu"
          onClick={[Function]}
        >
          <pure(Menu) />
        </WithStyles(IconButton)>,
        <WithStyles(Drawer)
          onClick={[Function]}
          open={false}
        >
          <Link
            replace={false}
            to="registration"
          >
            <WithStyles(MenuItem)
              className="nabi-color-nabi nabi-margin-top-small"
              onClick={[Function]}
            >
              Create Account
            </WithStyles(MenuItem)>
          </Link>
          <Link
            replace={false}
            to="info-parents"
          >
            <WithStyles(MenuItem)
              className={undefined}
              onClick={[Function]}
            >
              Information for parents
            </WithStyles(MenuItem)>
          </Link>
          <Link
            replace={false}
            to="info-instructors"
          >
            <WithStyles(MenuItem)
              className={undefined}
              onClick={[Function]}
            >
              Become a Nabi instructor
            </WithStyles(MenuItem)>
          </Link>
          <Link
            replace={false}
            to="info-instruments"
          >
            <WithStyles(MenuItem)
              className={undefined}
              onClick={[Function]}
            >
              Don't have an instrument?
            </WithStyles(MenuItem)>
            <WithStyles(Divider) />
          </Link>
          <Link
            replace={false}
            to="terms-of-use"
          >
            <WithStyles(MenuItem)
              className={undefined}
              onClick={[Function]}
            >
              Terms of use
            </WithStyles(MenuItem)>
          </Link>
          <Link
            replace={false}
            to="contact"
          >
            <WithStyles(MenuItem)
              className={undefined}
              onClick={[Function]}
            >
              Contact Us
            </WithStyles(MenuItem)>
          </Link>
        </WithStyles(Drawer)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-label": "Menu",
          "children": <pure(Menu) />,
          "id": "menu",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="registration"
            >
              <WithStyles(MenuItem)
                className="nabi-color-nabi nabi-margin-top-small"
                onClick={[Function]}
              >
                Create Account
              </WithStyles(MenuItem)>
            </Link>,
            <Link
              replace={false}
              to="info-parents"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Information for parents
              </WithStyles(MenuItem)>
            </Link>,
            <Link
              replace={false}
              to="info-instructors"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Become a Nabi instructor
              </WithStyles(MenuItem)>
            </Link>,
            <Link
              replace={false}
              to="info-instruments"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Don't have an instrument?
              </WithStyles(MenuItem)>
              <WithStyles(Divider) />
            </Link>,
            <Link
              replace={false}
              to="terms-of-use"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Terms of use
              </WithStyles(MenuItem)>
            </Link>,
            <Link
              replace={false}
              to="contact"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Contact Us
              </WithStyles(MenuItem)>
            </Link>,
          ],
          "onClick": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "createAccount",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(MenuItem)
                  className="nabi-color-nabi nabi-margin-top-small"
                  onClick={[Function]}
                >
                  Create Account
                </WithStyles(MenuItem)>,
                undefined,
              ],
              "replace": false,
              "to": "registration",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Create Account",
                  "className": "nabi-color-nabi nabi-margin-top-small",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Create Account",
                "type": [Function],
              },
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "infoParents",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Information for parents
                </WithStyles(MenuItem)>,
                undefined,
              ],
              "replace": false,
              "to": "info-parents",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Information for parents",
                  "className": undefined,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Information for parents",
                "type": [Function],
              },
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "infoInstructors",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Become a Nabi instructor
                </WithStyles(MenuItem)>,
                undefined,
              ],
              "replace": false,
              "to": "info-instructors",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Become a Nabi instructor",
                  "className": undefined,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Become a Nabi instructor",
                "type": [Function],
              },
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "infoInstruments",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Don't have an instrument?
                </WithStyles(MenuItem)>,
                <WithStyles(Divider) />,
              ],
              "replace": false,
              "to": "info-instruments",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Don't have an instrument?",
                  "className": undefined,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Don't have an instrument?",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "termsOfUse",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Terms of use
                </WithStyles(MenuItem)>,
                undefined,
              ],
              "replace": false,
              "to": "terms-of-use",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Terms of use",
                  "className": undefined,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Terms of use",
                "type": [Function],
              },
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "contact",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Contact Us
                </WithStyles(MenuItem)>,
                undefined,
              ],
              "replace": false,
              "to": "contact",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Contact Us",
                  "className": undefined,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Contact Us",
                "type": [Function],
              },
              undefined,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(IconButton)
            aria-label="Menu"
            id="menu"
            onClick={[Function]}
          >
            <pure(Menu) />
          </WithStyles(IconButton)>,
          <WithStyles(Drawer)
            onClick={[Function]}
            open={false}
          >
            <Link
              replace={false}
              to="registration"
            >
              <WithStyles(MenuItem)
                className="nabi-color-nabi nabi-margin-top-small"
                onClick={[Function]}
              >
                Create Account
              </WithStyles(MenuItem)>
            </Link>
            <Link
              replace={false}
              to="info-parents"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Information for parents
              </WithStyles(MenuItem)>
            </Link>
            <Link
              replace={false}
              to="info-instructors"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Become a Nabi instructor
              </WithStyles(MenuItem)>
            </Link>
            <Link
              replace={false}
              to="info-instruments"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Don't have an instrument?
              </WithStyles(MenuItem)>
              <WithStyles(Divider) />
            </Link>
            <Link
              replace={false}
              to="terms-of-use"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Terms of use
              </WithStyles(MenuItem)>
            </Link>
            <Link
              replace={false}
              to="contact"
            >
              <WithStyles(MenuItem)
                className={undefined}
                onClick={[Function]}
              >
                Contact Us
              </WithStyles(MenuItem)>
            </Link>
          </WithStyles(Drawer)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Menu",
            "children": <pure(Menu) />,
            "id": "menu",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="registration"
              >
                <WithStyles(MenuItem)
                  className="nabi-color-nabi nabi-margin-top-small"
                  onClick={[Function]}
                >
                  Create Account
                </WithStyles(MenuItem)>
              </Link>,
              <Link
                replace={false}
                to="info-parents"
              >
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Information for parents
                </WithStyles(MenuItem)>
              </Link>,
              <Link
                replace={false}
                to="info-instructors"
              >
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Become a Nabi instructor
                </WithStyles(MenuItem)>
              </Link>,
              <Link
                replace={false}
                to="info-instruments"
              >
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Don't have an instrument?
                </WithStyles(MenuItem)>
                <WithStyles(Divider) />
              </Link>,
              <Link
                replace={false}
                to="terms-of-use"
              >
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Terms of use
                </WithStyles(MenuItem)>
              </Link>,
              <Link
                replace={false}
                to="contact"
              >
                <WithStyles(MenuItem)
                  className={undefined}
                  onClick={[Function]}
                >
                  Contact Us
                </WithStyles(MenuItem)>
              </Link>,
            ],
            "onClick": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "createAccount",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(MenuItem)
                    className="nabi-color-nabi nabi-margin-top-small"
                    onClick={[Function]}
                  >
                    Create Account
                  </WithStyles(MenuItem)>,
                  undefined,
                ],
                "replace": false,
                "to": "registration",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Create Account",
                    "className": "nabi-color-nabi nabi-margin-top-small",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Create Account",
                  "type": [Function],
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "infoParents",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(MenuItem)
                    className={undefined}
                    onClick={[Function]}
                  >
                    Information for parents
                  </WithStyles(MenuItem)>,
                  undefined,
                ],
                "replace": false,
                "to": "info-parents",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Information for parents",
                    "className": undefined,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Information for parents",
                  "type": [Function],
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "infoInstructors",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(MenuItem)
                    className={undefined}
                    onClick={[Function]}
                  >
                    Become a Nabi instructor
                  </WithStyles(MenuItem)>,
                  undefined,
                ],
                "replace": false,
                "to": "info-instructors",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Become a Nabi instructor",
                    "className": undefined,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Become a Nabi instructor",
                  "type": [Function],
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "infoInstruments",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(MenuItem)
                    className={undefined}
                    onClick={[Function]}
                  >
                    Don't have an instrument?
                  </WithStyles(MenuItem)>,
                  <WithStyles(Divider) />,
                ],
                "replace": false,
                "to": "info-instruments",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Don't have an instrument?",
                    "className": undefined,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Don't have an instrument?",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "termsOfUse",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(MenuItem)
                    className={undefined}
                    onClick={[Function]}
                  >
                    Terms of use
                  </WithStyles(MenuItem)>,
                  undefined,
                ],
                "replace": false,
                "to": "terms-of-use",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Terms of use",
                    "className": undefined,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Terms of use",
                  "type": [Function],
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "contact",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(MenuItem)
                    className={undefined}
                    onClick={[Function]}
                  >
                    Contact Us
                  </WithStyles(MenuItem)>,
                  undefined,
                ],
                "replace": false,
                "to": "contact",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Contact Us",
                    "className": undefined,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Contact Us",
                  "type": [Function],
                },
                undefined,
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
